### PR TITLE
Record nodes access optimization

### PR DESCRIPTION
**Pull request by fefrain**

### CODE REVIEW

File: common/record/record.js [hunk 1]
```diff
@@ -212,16 +229,84 @@ const deleteNode = node =>
...
 
     return R.reduce(
       (recordCurrent, child) => deleteNode(child)(recordCurrent),
-      recordValidationUpdated,
+      recordCacheUpdated,
       children
     )
   }
 
+// ======= CACHE
```
Comment (minotogna) [inline common/record/record.js]: please, document the data structure and methods

File: common/record/record.js [hunk 2]
```diff
@@ -212,16 +229,84 @@ const deleteNode = node =>
...
+
```
Comment (minotogna) [inline common/record/record.js]: Record.js is becoming huge, would it make sense to move the cache part into a separate file, e.g. RecordCache or .. (that is accessible only from within record.js)?

File: server/modules/record/persistence/recordManager.js [hunk 3]
```diff
@@ -1,11 +1,12 @@
 const db = require('../../../db/db')
 
+const Record = require('../../../../common/record/record')
+const { toUuidIndexedObj } = require('../../../../common/survey/surveyUtils')
```
Comment (minotogna) [inline server/modules/record/persistence/recordManager.js]: should  we use  ```const SurveyUtils = require('../../../../common/survey/surveyUtils')``` ?

File: common/record/record.js (hunk 2 above)
Comment (fefrain) [reply common/record/record.js]: yes we can

File: common/record/recordCache.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
+const R = require('ramda')
```
Comment (minotogna) [inline common/record/recordCache.js]: Please, comment data structure and methods.

File: common/record/recordCache.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
```
Comment (minotogna) [inline common/record/recordCache.js]: should we move the file into _internal folder?

**Pull request merged**